lib/sentry/client_report.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Sentry.ClientReport do
+  @moduledoc """
+  **TODO** add proper doc for module
+
+  See <https://develop.sentry.dev/sdk/client-reports/>.
+  """
+
+  @moduledoc since: "10.8.0"
+
+  use GenServer
+  alias Sentry.{Client, Config}
+
+  @client_report_reasons [
+    :ratelimit_backoff,
+    :queue_overflow,
+    :cache_overflow,
+    :network_error,
+    :sample_rate,
+    :before_send,
+    :event_processor,
+    :insufficient_data,
+    :backpressure
+  ]
+
+  @typedoc """
+  The possible reasons of the discarded event.
+  """
+  @type reason() ::
+          unquote(Enum.reduce(@client_report_reasons, &quote(do: unquote(&1) | unquote(&2))))
+
+  @typedoc """
+  The struct for a **client report** interface.
+  """
+  @type t() :: %__MODULE__{
+          timestamp: String.t() | number(),
+          discarded_events: [%{reason: reason(), category: String.t(), quantity: pos_integer()}]
+        }
+
+  defstruct [:timestamp, discarded_events: %{}]
+
+  @send_interval 5000
+
+  @spec start_link([]) :: GenServer.on_start()
+  def start_link([]) do
```

Comment:
If you choose to go with `Sentry.ClientReport` containing both the struct as well as the GenServer, you'll need to make all these functions private (`@doc false`).

Reply:
Updated but double check that this is what you meant -> 

``` elixir
@doc false
  @spec start_link([]) :: GenServer.on_start()
  def start_link([]) do
    # check config to see if send_client_report is true
    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
  end
```